node/shard/src/lib.rs

```diff
@@ -146,30 +185,27 @@ impl ShardBlockChain {
             parent_block_hash: parent_hash,
             shard_id: block.body.header.shard_id,
         };
-        let apply_result = self.runtime.write().check(
+        let apply_result = self.runtime.write().apply(
             &apply_state,
             &[],
             &block.body.transactions,
         );
-        match apply_result {
-            Some((db_transaction, root, tx_result)) => {
-                if root != block.header().body.merkle_root_state {
-                    info!(
-                        "Merkle root {} is not equal to received {} after applying the transactions from {:?}",
-                        block.header().body.merkle_root_state,
-                        root,
-                        block
-                    );
-                    false
-                } else {
-                    self.insert_block(&block, db_transaction, tx_result);
-                    true
-                }
-            }
-            None => {
-                info!("Found incorrect transaction in block {:?}", block);
-                false
-            }
+        if apply_result.root != block.header().body.merkle_root_state {
```

Comment:
don't use `header()` unless you actually need signed header struct.
instead can juse use `block.body.header`

Reply:
So why do we have `SignedShardBlockHeader` as a separate thing?

Reply:
Fixed in 00f85e7